src/index.ts

```diff
@@ -62,6 +62,8 @@ function main(): void {
 
   readEnvironmentConfig();
 
+  initLogger();
```

Comment:
Since it's still useful to have some debug logs for some operations before we read the configs, I suggest we initialize the logger twice: as soon as possible (e.g. as it was before, at import time), and then also here, after reading the environment configuration. With this approach, you can still get debug outputs for early operations if you define `CRAFT_LOG_LEVEL=debug` in your environment manually.

Reply:
>With this approach, you can still get debug outputs for early operations if you define CRAFT_LOG_LEVEL=debug in your environment manually.

Is that how `consola` works? I wouldn't expect old messages to be displayed after initializing the logger. Also what should be the default logging value? Right now since this relies on a global that is being shared, logger is still active, just at the default logging level.

Reply:
I don't expect it to reprint the old messages, but if we run `initLogger` at the import time, it will allow us to see debug/trace output for `checkForUpdates`, `printVersion`, and `readEnvironmentConfig` functions, based on `CRAFT_LOG_LEVEL` that _might_ be present in the environment. 
Then, when we run `initLogger` here for the second time, it picks up the log level value from config files and uses it (if anything is there). 

The only non-idempotent thing in `initLogger` is  `consola.reporters.push(new SentryBreadcrumbReporter());`, but after it is fixed, we can call `initLogger` pretty much after every environment change, to make sure that we're using the latest `CRAFT_LOG_LEVEL` value.

Does this make sense?

Reply:
Ah, yeah it is clear now and makes total sense. Will make the updates.